pkgs/build-support/dotnet/build-dotnet-module/hooks/dotnet-build-hook.sh

```diff
@@ -1,70 +1,86 @@
-# inherit arguments from derivation
-dotnetBuildFlags=( ${dotnetBuildFlags[@]-} )
-
 dotnetBuildHook() {
     echo "Executing dotnetBuildHook"
 
     runHook preBuild
 
-    if [ "${enableParallelBuilding-}" ]; then
+    local -r hostRuntimeId=@runtimeId@
+    local -r dotnetBuildType="${dotnetBuildType-Release}"
+    local -r dotnetRuntimeId="${dotnetRuntimeId-$hostRuntimeId}"
+
+    if [[ -n $__structuredAttrs ]]; then
+        local dotnetProjectFilesArray=( "${dotnetProjectFiles[@]}" )
+        local dotnetTestProjectFilesArray=( "${dotnetTestProjectFiles[@]}" )
+        local dotnetFlagsArray=( "${dotnetFlags[@]}" )
+        local dotnetBuildFlagsArray=( "${dotnetBuildFlags[@]}" )
+    else
+        local dotnetProjectFilesArray=($dotnetProjectFiles)
+        local dotnetTestProjectFilesArray=($dotnetTestProjectFiles)
+        local dotnetFlagsArray=($dotnetFlags)
+        local dotnetBuildFlagsArray=($dotnetBuildFlags)
+    fi
+
+    if [[ -n "${enableParallelBuilding-}" ]]; then
         local -r maxCpuFlag="$NIX_BUILD_CORES"
         local -r parallelBuildFlag="true"
     else
         local -r maxCpuFlag="1"
         local -r parallelBuildFlag="false"
     fi
 
-    if [ "${selfContainedBuild-}" ]; then
-        dotnetBuildFlags+=("-p:SelfContained=true")
+    if [[ -n ${dotnetSelfContainedBuild-} ]]; then
```

Comment:
There's a lot of renaming here that I don't necessarily think is an improvement, can we remove that at least for the sake of simplifying this change?

Reply:
Sure, I’ll move variable renaming to a separate commit. FWIW renaming these variables makes them clearly dotnet-specific to avoid potential collisions with other setup hooks. But, yes, this should be in a separate change from just fixing structured attributes support.

Reply:
> commit. FWIW renaming these variables makes them clearly dotnet-specific to avoid potential collisions with other setup hooks.

Agreed for global variables, but maybe not for locals?  I'm also not a big fan of using 'Array' as a suffix, because it's not a common convention afaict.

Reply:
Structured attributes are a relatively new feature, so a lot of things in Nixpkgs don’t explicitly support them. Stdenv does, but language frameworks that wrap it like .NET and Go build infrastructure usually break in completely unexpected ways.

I can think of one example that does similarly to this PR:
https://github.com/NixOS/nixpkgs/blob/16b7680853d2d0c7a120c21266eff4a2660a3207/pkgs/build-support/setup-hooks/auto-patchelf.sh#L56-L66

That said, I don’t have strong opinion on any particular code style w.r.t. implementation, so having different, shorter name for locals is fine as well. I’ll try to move as much renaming as possible to separate commit.